src/braket/aws/aws_emulator_helpers.py

```diff
@@ -0,0 +1,200 @@
+from collections.abc import Iterable
+from functools import singledispatch
+from typing import Union
+
+from networkx import DiGraph
+
+from braket.device_schema import DeviceActionType, DeviceCapabilities
+from braket.device_schema.ionq import IonqDeviceCapabilities
+from braket.device_schema.iqm import IqmDeviceCapabilities
+from braket.device_schema.rigetti import RigettiDeviceCapabilities
+from braket.emulators.emulator_passes import (
+    ConnectivityCriterion,
+    GateConnectivityCriterion,
+    GateCriterion,
+    QubitCountCriterion,
+)
+
+
+def create_qubit_count_criterion(properties: DeviceCapabilities) -> QubitCountCriterion:
+    """
+    Create a QubitCountCriterion pass which checks that the number of qubits used in a program does
+    not exceed the number of qubits allowed by a QPU, as defined in the device properties.
+
+    Args:
+        properties (DeviceCapabilities): QPU Device Capabilities object with a
+            QHP-specific schema.
+
+    Returns:
+        QubitCountCriterion: An emulator pass that checks that the number of qubits used in a
+        program does not exceed that of the max qubit count on the device.
+    """
+    qubit_count = properties.paradigm.qubitCount
+    return QubitCountCriterion(qubit_count)
+
+
+def create_gate_criterion(properties: DeviceCapabilities) -> GateCriterion:
+    """
+    Create a GateCriterion pass which defines what supported and native gates are allowed in a
+    program based on the provided device properties.
+
+    Args:
+        properties (DeviceCapabilities): QPU Device Capabilities object with a
+            QHP-specific schema.
+
+    Returns:
+        GateCriterion: An emulator pass that checks that a circuit only uses supported gates and
+        verbatim circuits only use native gates.
+    """
+
+    supported_gates = properties.action[DeviceActionType.OPENQASM].supportedOperations
+    native_gates = properties.paradigm.nativeGateSet
+
+    return GateCriterion(supported_gates=supported_gates, native_gates=native_gates)
+
+
+@singledispatch
```

Comment:
`singledispatch` produces really ugly documentation when used on public functions; only use it on private methods, like so:
```python
def connectivity_criterion(...):
    return _connectivity_criterion(...)

@singledispatch
def _connectivity_criterion(...):
    ...
```

Reply:
Updated! Thank you for this callout, I wasn't aware of that interaction!